src/rgw/rgw_rados.cc

```diff
@@ -121,17 +121,28 @@ static string RGW_DEFAULT_PERIOD_ROOT_POOL = "rgw.root";
 static bool rgw_get_obj_data_pool(const RGWZoneGroup& zonegroup, const RGWZoneParams& zone_params,
                                   const string& placement_id, const rgw_obj& obj, rgw_pool *pool)
 {
-  if (!zone_params.get_head_data_pool(placement_id, obj, pool)) {
-    RGWZonePlacementInfo placement;
-    if (!zone_params.get_placement(zonegroup.default_placement, &placement)) {
-      return false;
-    }
+  bool find;
+
+  if (obj.is_head_obj())
```

Comment:
@fangyuxiangGL seems to me that you'd want to audit the callers, and see whether they need to access the head or the tail and change behavior accordingly.